eras/conway/impl/src/Cardano/Ledger/Conway/Rules/HardFork.hs

```diff
@@ -74,7 +74,11 @@ hardforkTransition = do
         epochState
           & esLStateL . lsCertStateL %~ \certState ->
             let umap = certState ^. certDStateL . dsUnifiedL
-                dReps = certState ^. certVStateL . vsDRepsL
+                dReps =
+                  -- Reset all delegations in order to remove any inconsistencies
+                  -- Delegations will be reset accordingly below.
+                  Map.map (\dRepState -> dRepState {drepDelegs = Set.empty}) $
+                    certState ^. certVStateL . vsDRepsL
```

Comment:
There's a lens for `drepDelegs` so you could write this as

```haskell
Map.map (drepDelegsL .~ Set.empty) $
  certState ^. certVStateL . vsDRepsL
```

or even (if you want to be fancy)

```haskell
certState ^. certVStateL . vsDRepsL & traversed . drepDelegsL .~ Set.empty
```

However, no problem leaving it as it is if you want to get this finished sooner.